doc/mysqlnd_azure_log.md

```diff
@@ -0,0 +1,124 @@
+# Diagnostic Log Configuration
+
+Log down some runtime information from mysqlnd\_azure.
+
+## Configuration
+
```

Comment:
Add a single line here on what this section is about

Reply:
Added. I also refactor the explanation details section for better perception, and for test scenario check. Can you help have another look too?